src/test/java/org/swisspush/redisques/DequeueStatisticCollectorTest.java

```diff
@@ -0,0 +1,83 @@
+package org.swisspush.redisques;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.shareddata.AsyncMap;
+import io.vertx.core.shareddata.SharedData;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.swisspush.redisques.util.DequeueStatistic;
+import org.swisspush.redisques.util.DequeueStatisticCollector;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(VertxUnitRunner.class)
+public class DequeueStatisticCollectorTest {
+
+    private Vertx vertx;
+    private SharedData sharedData;
+    private AsyncMap<String, DequeueStatistic> asyncMap;
+    private DequeueStatisticCollector dequeueStatisticCollectorEnabled;
+    private DequeueStatisticCollector dequeueStatisticCollectorDisabled;
+
+    @Before
+    public void setUp() {
+        vertx = mock(Vertx.class);
+        sharedData = mock(SharedData.class);
+        asyncMap = mock(AsyncMap.class);
+
+        doAnswer(invocation -> {
+            io.vertx.core.Handler<io.vertx.core.AsyncResult<AsyncMap<String, DequeueStatistic>>> handler = invocation.getArgument(1);
+            handler.handle(Future.succeededFuture(asyncMap));
+            return null;
+        }).when(sharedData).getAsyncMap(anyString(), any());
+
+        when(vertx.sharedData()).thenReturn(sharedData);
+
+        // Set up the enabled and disabled DequeueStatisticCollector
+        dequeueStatisticCollectorEnabled = new DequeueStatisticCollector(vertx, true);
+        dequeueStatisticCollectorDisabled = new DequeueStatisticCollector(vertx, false);
+    }
+
+    @Test
+    public void testGetAllDequeueStatisticsEnabled(TestContext context) {
```

Comment:
Good point to create a unit test for the `DequeueStatisticCollector`.  Since you only worked with the `getAllDequeueStatistics()` method it is ok to write tests for this method only.

However, you should test all cases, not only the "happy cases". What I mean is in the `getAllDequeueStatistics()` method there are two async calls to `sharedData` and `asyncMap` which can fail. You should write tests for these cases too.

Reply:
Sure, I will create 2 more failure tests to cover these scenarios.